plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -671,7 +671,7 @@ private Iterator<InternalHiveSplit> createInternalHiveSplitIterator(TrinoFileSys
     private static void checkPartitionLocationExists(TrinoFileSystem fileSystem, Location location)
     {
         try {
-            if (!fileSystem.directoryExists(location).orElse(true)) {
+            if (!fileSystem.directoryExists(location).orElse(!ignoreAbsentPartitions)) {
```

Comment:
Could you add a test? The test should fail without this change and pass with this change.

Reply:
In addition, is this the right place to add this change?
This is a static method and so if you try to refer the non-static variable, compiler generates an error.

The variable is checked before calling the method in 2 out of 3 locations:

[https://github.com/trinodb/trino/blob/33cb3ad06485c403e0f9bc322f7d28642fbc92d1/plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java#L506](https://github.com/trinodb/trino/blob/33cb3ad06485c403e0f9bc322f7d28642fbc92d1/plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java#L506)

Looks like this needs to be fixed in this block:
[https://github.com/trinodb/trino/blob/49e42f8116a35cdb76138b69b7d4aecad56248b1/plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java#L532](https://github.com/trinodb/trino/blob/49e42f8116a35cdb76138b69b7d4aecad56248b1/plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java#L532)